omnigibson/systems/macro_particle_system.py

```diff
@@ -847,7 +947,10 @@ def _dump_state(cls):
                 particle_attached_obj_uuid=cls._group_objects[group_name].uuid,
                 n_particles=cls.num_group_particles(group=group_name),
                 particle_idns=[cls.particle_name2idn(name=name) for name in group_particles.keys()],
-                particle_attached_link_names=[cls._particles_info[name]["link"].prim_path.split("/")[-1] for name in group_particles.keys()],
+                particle_attached_link_names=[
+                    # If the attached object is a cloth, store the negative of the face_id instead!
+                    -cls._particles_info[name]["face_id"] if cls._particles_info[name]["link"] is None else
```

Comment:
sorry not sure I understand this either.

Reply:
bad code, fixed